### DISCUSSION TITLE
Full content of Message the quote refers to

### DISCUSSION THREAD

**Discussion by MeinDeutschkurs** [tulir/whatsmeow#391] [WhatsApp protocol Q&A] [answered]
Hello everyone,

I apologize for the newbie question.

I have a situation where I need to distinguish between a quoted message and an initial message. Here's the code snippet I'm currently using:

```
var msg string
if quoted == nil {
  msg = newMessage.GetConversation()
} else {
  msg = quoted.GetText()
}
```

Now, I would like to know if there's a way to retrieve the original message that the quote refers to. I want to use it as a form of context.

I truly appreciate all of your work, assistance and support. Thank you so much,
Johannes

**Comment by hoehermann** [comment, marked as answer]
When receiving a quote, the quoted message is given in the [ContextInfo](https://godocs.io/go.mau.fi/whatsmeow/binary/proto#ContextInfo). Here is an example how I use it: https://github.com/hoehermann/purple-gowhatsapp/blob/60b7398/src/go/handle_message.go#L39

**Reply by MeinDeutschkurs** [reply]
Thank you so much, @hoehermann !
